### DISCUSSION TITLE
Dropdown doesn't update after setValue()

### DISCUSSION THREAD

**Discussion by seabasss** [orchidjs/tom-select#179] [Q&A]
I update the value of my single value tom select based on a value from a different select. I use `setValue()` to make this happen and everything looks fine, until I open the dropdown and the old value is still selected in the list of options. I added a `clearCache()` after it and it seems to partly fix it, until I start typing and both the old and the new values are selected in the dropdown list. What's the proper way of doing this? Is there a search cache that needs to be cleared as well?

`window.myTomSelect.setValue(event.detail.updatedTimezone);
window.myTomSelect.clearCache();`

Thank you!

**Comment by oyejorge** [comment]
Can you put together a small example on jsfiddle (or similar)?

**Comment by seabasss** [comment]
I think this is my bad. When you start searching it selects the best match, which made it look like everything I searched for was selected. Doh. Sorry.

However, it seems like I still need `clearCache()` after `setValue()` for the selected value to also be selected in the dropdown (before searching!!). Is this the correct way to do it?
